File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.java
class Solution {
    public int peakIndexInMountainArray(int[] arr) {
        int l = 0, r = arr.length-1;
        while(l < r) {   // The equal interfered this time
            int m = l + (r-l)/2;
            if (arr[m] < arr[m+1]) {
                l = m + 1;
            } else {
                r = m;
            }
        }
        return l;
    }
}

File: 852-peak-index-in-a-mountain-array/852-peak-index-in-a-mountain-array.kt
class Solution {
    fun peakIndexInMountainArray(arr: IntArray): Int { 
        if (arr.size < 3) return -1
        var left = 0; var right = arr.lastIndex
        while (left <= right) {
            val mid = left + (right-left)/2
            if (mid-1 >= 0 && mid+1 <= arr.lastIndex &&
                arr[mid-1] < arr[mid] && arr[mid] > arr[mid+1]) {
                return mid 
            } else if (mid-1 >= 0 && arr[mid-1] >= arr[mid]) {
                right = mid - 1
            } else {
                left = mid + 1
            }
        }
        return -1
    }
}